sound/Scale.java

```diff
@@ -6,14 +6,34 @@
 /**
  * A musical scale.
  * Comprised of an ordered set of pitches (midi note numbers).
- * 
+ *
  * @author Dr. Jody Paul
  * @version 1.2
  */
 public class Scale {
     /** The pitches for this tune. */
     private List<Integer> scalePitches;
-    
+    /** Start of pitchInterval.*/
+    private static final int START_PITCH_INTERVAL = -3;
+    /** End of pitchInterval.*/
+    private static final int END_PITCH_INTERVAL = -3;
+    /** The factor to move between octaves.*/
+    private static final int  OCTAVE_FACTOR = 12;
+    /** A-Sharp in each octave.*/
+    private static final int A_SHARP = 10;
+    /** Natural A in each octave.*/
+    private static final int A_NAT = 9;
+    /** Natural G in each octave.*/
+    private static final int G_NAT = 7;
+    /** Natural F in each octave.*/
+    private static final int F_NAT = 5;
+    /** Natural E in each octave.*/
+    private static final int E_NAT = 4;
+    /** D-Sharp in each octave.*/
+    private static final int D_SHARP = 3;
+    /** Natural D in each octave.*/
+    private static final int D_NAT = 2;
```

Comment:
These are intended to be interval values, not pitches.  Thus D_NAT is better labeled as "MAJOR_SECOND".  Reading **up** from there, the intervals would be MINOR_THIRD, MAJOR_THIRD, PERFECT_FOURTH, PERFECT_FIFTH, MAJOR_SIXTH, MINOR_SEVENTH.

Reply:
Who knew I could learn music theory and software development in the same class!

Reply:
Also, regarding JavaDoc's on these should I describe them in terms of the amount of half steps of that interval or just replace the original pitch name with the new interval name?

Reply:
> Who knew I could learn music theory and software development in the same class!

Yes, indeed!  We generally need to learn more of the domain than we expected to.   But that makes sense, given that programs essentially embody models of the domain.

What's particularly useful about this specific case is that it demonstrate just how meaningless the "magic numbers" were.  Only once we tried to associate meaning (semantics) with them--by using symbolic constants--did we discover what those magic numbers were actually intended to mean.

Reply:
> Also, regarding JavaDoc's on these should I describe them in terms of the amount of half steps of that interval or just replace the original pitch name with the new interval name?

You pose an interesting question.  The answer depends on what we can expect future maintainers to know and what information is useful to them.  I think it may be worth it, if it can be kept short, to include sufficient information for the integer value to make sense.  For example, if we think that "half-steps" is understood, `/** Number of half-steps in a minor third interval. */`  I'm wondering if replacing "half-septs" by something like the _number of notes_ would be understandable to a broader set of coders.

Reply:
It looks like the current number specifies the number of semitones in that interval. I believe the number of notes in a semitone is double the value of the semitone? It may be confusing to refer to the number of notes in the interval if that is the case. Should I just have the JavaDoc reference the number of semitones?

Reply:
"Notes" is very generic and imprecise.  I agree with the use of "semintones".